File: server/storage/src/main/java/ru/storage/person/Position.java
package ru.storage.person;

/**
 * Enum defining the user type folder where user pictures are stored in S3 bucket (e.g. "director", "actor").
 */
public enum Position {

    DIRECTOR("DIRECTOR"),
    ACTOR("ACTOR"),
    OPERATOR("OPERATOR"),
    PRODUCER("PRODUCER");

    public final String stringValue;

    Position(String stringValue)  {
        this.stringValue = stringValue;
    }

}

File: web/src/types/Position.ts
enum Position {
    DIRECTOR = 'DIRECTOR',
    ACTOR = 'ACTOR',
    PRODUCER = 'PRODUCER',
    OPERATOR = 'OPERATOR',
}

const PositionKindLocalized = {
    RU: {
        [Position.DIRECTOR]: 'Режиссёр',
        [Position.ACTOR]: 'Актёр',
        [Position.PRODUCER]: 'Продюсер',
        [Position.OPERATOR]: 'Оператор',
    },
    EN_FROM_RU: {
        ['Режиссёр']: Position.DIRECTOR,
        ['Актёр']: Position.ACTOR,
        ['Продюсер']: Position.PRODUCER,
        ['Оператор']: Position.OPERATOR,
    },
}

export default Position
export { PositionKindLocalized }
